backend/corpora/dataset_processing/download.py

```diff
@@ -9,6 +9,7 @@
     from ..common.entities import Dataset
     from ..common.utils.db_utils import db_session_manager
     from ..common.utils.math_utils import MB
+    from ..lambdas.upload_failures.upload import delete_many_from_s3
```

Comment:
if `delete_many_from_s3` will be used by multiple deployed resources then it should live in `common`. I think in `common.utils` would be a good start. This helps us avoid circular dependencies and keeps the code's hierarchy in tact.